### PR TITLE
[file_selector] Endorse the web implementation

### PR DESCRIPTION
**Pull request by tugorez**
This PR endorses [file_selector_web][1] as [file_selector][2]'s web implementation.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

[1]: https://pub.dev/packages/file_selector_web
[2]: https://pub.dev/packages/file_selector

### CODE REVIEW

**Comment by tugorez** [comment]
cc @ditman

**Comment by ditman** [comment]
Before this gets merged, can you please update https://github.com/flutter/plugins/tree/master/packages/file_selector/file_selector_platform_interface so it depends in the latest version of `cross_file`?

**Comment by tugorez** [comment]
> Before this gets merged, can you please update https://github.com/flutter/plugins/tree/master/packages/file_selector/file_selector_platform_interface so it depends in the latest version of `cross_file`?

Sure, already sent https://github.com/flutter/plugins/pull/3422. After it is release I'll make the ammendments and change this PR's status to ready for review. Thanks @ditman :)

**Comment by ditman** [comment]
I've just merged #3422, now waiting for CI to publish it.

**Comment by ditman** [comment]
The failing tests here were my fault, we never published 1.0.3 of the platform_interface. I've now published 1.0.3+1 and restarted the tests here. Everything should pass now.

**Review by ditman — APPROVED** [review]

**Comment by ditman** [comment]
@googlebot I consent.

**Comment by ditman** [comment]
This will need to be rebased again, it's missing a CI fix from master! Sorry for the inconvenience!

**Comment by tugorez** [comment]
This PR is ready to go @ditman :)!

**Comment by tugorez** [comment]
This was already endorsed https://github.com/flutter/plugins/blob/master/packages/file_selector/file_selector/pubspec.yaml. Marking this as closed

**Pull request closed without merging**
